framework/auth/views.py

```diff
@@ -245,13 +245,8 @@ def auth_login(auth, **kwargs):
         next_url = request.args.get('redirect_url')
         must_login_warning = False
 
-    if next_url:
-        # Only allow redirects which are relative root or full domain, disallows external redirects.
-        if not (next_url[0] == '/'
-                or next_url.startswith(settings.DOMAIN)
-                or next_url.startswith(settings.CAS_SERVER_URL)
-                or next_url.startswith(settings.MFR_SERVER_URL)):
-            raise HTTPError(http.InvalidURL)
+    if not validate_next_url(next_url):
+        raise HTTPError(http.BAD_REQUEST)
```

Comment:
For Reviewer: OSF cannot handle `http.InvalidURL`, change to `http.BAD_REQUEST` which is more appropriate in this case.